api/app/auth.py

```diff
@@ -13,19 +15,38 @@
 
 
 async def authenticate(token: str = Depends(oauth2_scheme)):
-    """ Returns True when validation of the token is successful """
+    """ Returns True when validation of the token is successful. """
     # RSA public key format
     keycloak_public_key = '-----BEGIN PUBLIC KEY-----\n' + \
         config.get('KEYCLOAK_PUBLIC_KEY') + '\n-----END PUBLIC KEY-----'
 
     try:
-        jwt.decode(token, keycloak_public_key, algorithm='RS256')
-        return True
-    except Exception as exception:
-        detail = 'Could not validate the credential ({})'.format(exception)
+        decoded_token = jwt.decode(token, keycloak_public_key, algorithm='RS256')
+        return decoded_token
+    except InvalidTokenError:
+        detail = 'Could not validate the credential (Not enough segments)'
         logger.error(detail)
+        return {}
+
+
+async def audit(request: Request, token=Depends(authenticate)):
+    """ Audits attempted requests based on bearer token. """
+    path = request.url.path
+
+    if not token:
+        create_api_access_audit_log(str(None), False, path)
```

Comment:
str(None) creates a string that says 'None'. We don't want that - if the username is None, we want it to be None, not a string.
```suggestion
        create_api_access_audit_log(None, False, path)
```